### ISSUE TITLE
[Label] Wrapped components and focus shifting

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#8038]
**monday.com sync:** #18005777747

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

As identified in https://github.com/Esri/calcite-design-system/issues/5070#issuecomment-1760407588, certain browsers (Chrome, Safari, and Edge) shift focus to the first element when wrapped in `label`. Also applies to the [Charts team](https://github.com/Esri/calcite-design-system/issues/5070#issuecomment-1448632689).

cc @zaramatheson @AdelheidF @jcfranco 

![HoldnRelease](https://user-images.githubusercontent.com/50747646/182474711-5ad5b7bc-00f0-40b9-a993-4ac95e73ffde.gif)

### Expected Behavior

The behavior is similar to what is observed in native HTML, but there may be a solution that can be explored as Calcite's code is managing the focus behavior.

### Reproduction Sample

https://codepen.io/geospatialem/pen/poGvgrO

### Reproduction Steps

1. Open the [sample](https://codepen.io/geospatialem/pen/poGvgrO)
2. Click the `input` and hold the mouse button while moving the mouse over the `slider` and release the mouse button.
3. Depending on the browser, observe the following:
   - FF: Focus is not shifted
    - Chrome/Edge/Safari: When doing [this workflow](https://github.com/Esri/calcite-design-system/issues/5070#issuecomment-1760407588) the first element, in this case the `slider` shifts focus
4. Observe similar behavior on native elements

### Reproduction Version

1.9.2

### Relevant Info

#5070 addressed when elements are wrapped in a `calcite-label` that the first element should receive, or steal, focus.

### Priority impact

p3 - want for upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Map Viewer

**Comment by geospatialem** [comment]
There is a workaround, if it fits the use cases for folks, where each element can have its own individual label using the `label` attribute, also depicted in https://codepen.io/geospatialem/pen/poGvgrO with:

```html
<calcite-label for="my-slider">my label</calcite-label>
  <div>
    <calcite-slider id="my-slider"></calcite-slider>
    <calcite-input label="my label"></calcite-input>
  </div>
```

**Comment by macandcheese** [comment]
Is this valid anymore cc @Esri/calcite-pes

**Comment by DintaMel** [comment]
I couldn't recreate this in Safari, Firefox, or Chrome. I don't have Edge, so I couldn't test that.

**Comment by AdelheidF** [comment]
Still reproducible with 5.1 in Chrome.
Click into the 3rd input box. Then keep the mouse pressed, move up and release the mouse over the slider. Now the slider has the focus.

https://codepen.io/editor/afreitag/pen/01a08d44-d5db-7aef-b03f-c9d3ab89b4f2

**Comment by macandcheese** [comment]
Thanks for confirming - a few thoughts... I was under the impression multiple components wrapped in a single label was an a11y issue? We also have been moving away from the label component towards our new, built-in `label-text` props. It could also be seen as expected as it matches the native implementation...

**Comment by isaacbraun** [comment]
Agreed, I also believe wrapping multiple components in one `Label` is not recommended, but rather a `fieldset` wrapping multiple components that each have their own `label`. Based on that alone I feel like this could be closed. 🤔

**Comment by geospatialem** [comment]
> I was under the impression multiple components wrapped in a single label was an a11y issue?

> I also believe wrapping multiple components in one Label is not recommended, but rather a fieldset wrapping multiple components that each have their own label. Based on that alone I feel like this could be closed. 🤔

Agreed on both points. 

@AdelheidF we are aiming for a new component _soon_ that would allow more configuration options for the team. Aiming to close this one in favor of that effort to follow a11y best practices with additional similar support to native forms with https://github.com/Esri/calcite-design-system/issues/12450. 

If that sounds good to folks I can close this one out in favor of it.
